wpengine/faustjs
# Show Faust Toolbar based on user preferences in WordPress

## Tasks


## Description

This PR shows or hides the Faust Toolbar based on the user's toolbar preferences in WordPress:
<img width="599" alt="Screenshot 2023-03-31 at 11 13 15 AM" src="https://user-images.githubusercontent.com/5946219/229174203-7e13acfc-2723-4738-b041-053df46db2d0.png">

A field was added to the `viewer` model in WPGraphQL called `shouldShowFaustToolbar` which we use to either show or hide the toolbar component.

If the user is using a version of Faust where this field does not exist, the query will throw an error, and show the toolbar. If the query fails, there is no blocking errors, the behavior just falls back to showing the toolbar.



## Related Issue(s):



## Testing

Tests have been provided. Additionally, here are some manual testing instructions:

1. Checkout this branch
2. Build the core and cli packages
3. In the example app, ensure `experimentalToolbar` is set
4. Make sure you are using the version of FaustWP from this branch
5. Start the dev server
6. Authenticate via previews
7. Notice the toolbar will now show based on your preferences in WordPress



## Screenshots



## Documentation Changes



## Dependant PRs
